### PR TITLE
Migrate CosmWasm workspace

### LINKED ISSUE #1311: Split out `cosmwasm` workspace into new repository
## Summary

We recently separated the CosmWasm crates into their own workspace to decouple them from the ibc-rs dependencies. This was necessary because the coupling was causing conflicts and complicating the management of MSRV for the ibc-rs libraries.

However, we've encountered some challenges in our release process, publishing crates, and tracking changes across multiple workspaces. To simplify maintenance, it's more practical to move the `cosmwasm` directory into a separate repository.

Additionally, the versioning of these two workspaces doesn't need to be synchronized. As we prepare for the ibc-rs v1 release, the CosmWasm integration might not yet be ready for a production release.

### PR DESCRIPTION
**Pull request by seanchen1991**
Closes: #1311 

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -69,19 +69,16 @@ The release process for the main ibc-rs workspace is as follows:
       than 5 new crates or 30 existing crates by default. If we need to publish
       more than these limits, we need to update `release.toml` at workspace
       root.
-5. Run `cargo doc -p ibc --all-features --open` locally to double-check that all
```
Comment (seanchen1991) [inline RELEASES.md:72]: Removed this step as there is already a `make check-docs` rule that is specified in the `rust.yaml` file.

**Comment by seanchen1991** [comment]
Closing this PR as we didn't move the `cosmwasm` directory out in such a way that preserved its git history.

**Comment by seanchen1991** [comment]
On second thought, the commits in this PR should be fine to merge, as migrating the files out of ibc-rs should not have any adverse effects on its git history.

File: ibc-clients/README.md [hunk 2]
```diff
@@ -25,7 +25,6 @@ IBC light clients:
 
 - [ibc-client-tendermint-types](./ics07-tendermint/types): Data Structures
 - [ibc-client-tendermint](./ics07-tendermint): Implementation
-- [ibc-client-tendermint-cw](./../cosmwasm/ibc-clients/ics07-tendermint): CosmWasm Contract
```
Comment (Farhad-Shabani) [inline ibc-clients/README.md:28]: Let's add something here instead and bring this to attention that the implementation of ICS-07 CosmWasm contract lives under the cosmwasm-ibc repository. And include a link too.

File: README.md [hunk 3]
```diff
@@ -38,7 +38,6 @@ sub-crates that can be used independently or together.
 | [ibc-testkit](ibc-testkit)           | Provides testing toolkit to aid `ibc-rs` and host chains in writing integration tests.               |
 | [ibc-query](ibc-query)               | Contains utility traits and implementations for querying states of an integrated IBC module.         |
 | [ibc-derive](ibc-derive)             | Derive macros for `ClientState` and `ConsensusState` traits, reducing boilerplate.                   |
-| [cosmwasm](cosmwasm)                 | Contains libraries that implement `ibc-rs` modules as CosmWasm contracts.                            |
```
Comment (Farhad-Shabani) [inline README.md:41]: Can we add this project now under the `ibc-rs in the wild` section?

**Review by Farhad-Shabani — CHANGES_REQUESTED** [review]
Left a few requests. Please recored a changelog too.

I am also thinking it would be good as well to inform teams in the shared-community-ibc-rs about this new repo.

File: README.md (hunk 3 above)
Comment (seanchen1991) [reply README.md:41]: [32608ee](https://github.com/cosmos/ibc-rs/pull/1315/commits/32608eef115ba378a160a7c40c0549510fd450ca)

File: ibc-clients/README.md (hunk 2 above)
Comment (seanchen1991) [reply ibc-clients/README.md:28]: [32608ee](https://github.com/cosmos/ibc-rs/pull/1315/commits/32608eef115ba378a160a7c40c0549510fd450ca)

**Review by Farhad-Shabani — APPROVED** [review]
:pray:

**Pull request merged**